### DISCUSSION TITLE
Experiment stuck when 100% training completed 📡 

### DISCUSSION THREAD

**Discussion by daviddelriod** [securefederatedai/openfederatedlearning#760] [Q&A] [answered]
I have successfully started an FL experiment using the **interactive api** and all seems to work perfectly but when the training of the collaborators is finished, the experiment gets stuck. Simply, it gets there and it does not perform any agreggation or additional operation.

![completed](https://user-images.githubusercontent.com/64251001/226203247-9fc0f809-206e-4fb9-8bd8-c986a8c3ea10.png)

I also try to see the metrics via notebook but it does not throw any progress bar or metric in the output:

![note](https://user-images.githubusercontent.com/64251001/226203324-c1e195b6-9648-4580-899a-ac1e7272cd1c.png)

The experiment is defined like this: 

`fl_experiment.start(
    model_provider=model_interface, 
    task_keeper=task_interface,
    data_loader=fed_dataset,
    rounds_to_train=5,
    opt_treatment='CONTINUE_GLOBAL'
)`

Am I missing a task for the **aggregation function** or something related? (I have the same configuration as [tinyimage](https://github.com/securefederatedai/openfl/blob/develop/openfl-tutorials/interactive_api/PyTorch_TinyImageNet/workspace/pytorch_tinyimagenet.ipynb)'s example.

Thanks in advance!!

**Comment by daviddelriod** [comment, marked as answer]
The problem was that I was putting the **tqdm progress bar** for the **dataloader** and not for the **loop of epochs** so it just took into account the first iteration and the other epochs were executing "in the background". As shown in the image, the bar of training is in 100%. However, it remained 7 epochs yet as just one epoch (one complete iteration of the dataloader) had been trained.
